packages/e2e-contract-tests/playwright/e2e/MintAsset.test.ts

```diff
@@ -42,4 +44,68 @@ test.describe('Mint Assets', () => {
     // 0.000000001 since we hardcode a gas price of 1 in our mint function
     await hasText(walletPage, '0.000000001 ETH');
   });
+
+  test('e2e mint known asset', async ({ context, page }) => {
+    const connectButton = getButtonByText(page, 'Connect');
+    await connectButton.click();
+    await walletConnect(context);
+
+    const mintAmount = '1.2345';
+    const mintAmountInput = page.getByLabel('Asset config amount');
+    await mintAmountInput.fill(mintAmount);
+
+    const subId =
+      '0x0000000000000000000000000000000000000000000000000000000000000001';
+    const subIdInput = page.getByLabel('Asset config subid');
+    await subIdInput.fill(subId);
+
+    const decimals = '6';
+    const decimalsInput = page.getByLabel('Asset config decimals');
+    await decimalsInput.fill(decimals);
+
+    const mintButton = getButtonByText(page, 'Mint Asset configuration');
+    await page.waitForTimeout(3000);
+    await mintButton.click();
+
+    // test asset is correct
+    const assetId = calculateAssetId(process.env.VITE_CONTRACT_ID!, subId);
+    let walletPage = await getWalletPage(context);
+    // short address function copied from app package
+    await hasText(walletPage, shortAddress(assetId), 0, 10000);
+
+    // test mint amount is correct
+    await hasText(walletPage, '0.0012345');
+
+    // test gas fee is shown and correct
+    await hasText(walletPage, 'Fee (network)');
+    // 0.000000001 since we hardcode a gas price of 1 in our mint function
+    await hasText(walletPage, '0.000000001 ETH');
+
+    await walletApprove(context);
+
+    const name = 'Token';
+    const symbol = 'TKN';
+    // Add the unknown asset to the wallet
+    await addAsset(context, assetId, name, symbol, Number(decimals));
```

Comment:
can we do the `addAsset` before sending the contract transaction? on this way we can validate both screens with same asset configuration

Reply:
what's the point of validating the first screen with unknown asset and incorrect decimal config? and then after approve validate with the asset added ?

Reply:
my thinking for validating the unknown screen is ensuring that the wallet doesn't show any asset data before the asset is added

Reply:
the first test already validates the scenario for unknown asset... this new test should validating the scenario for known asset